Proteomics/ProteomicsExtenstionMethods.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Proteomics
+{
+    public static class ProteomicsExtenstionMethods
+    {
+        public static List<int> AllIndexesOf(this string str, string value)
+        {
+            if (String.IsNullOrEmpty(value))
+                throw new ArgumentException("the string to find may not be empty", "value");
+            List<int> indexes = new List<int>();
+            for (int index = 0; ; index += value.Length)
+            {
+                index = str.IndexOf(value, index);
+                if (index == -1)
+                    return indexes;
+                indexes.Add(index);
+            }
+        }
+
+        public static void Fill<T>(this T[] originalArray, T with)
```

Comment:
This method already exists?
https://docs.microsoft.com/en-us/dotnet/api/system.array.fill?view=netstandard-2.1